### PR TITLE
fix: issue #310

### PR DESCRIPTION
**Pull request by simonpfeifhofer**
Closes: #310 

## Description

- fix make target build-docker-local-ethermint
- fix make target localnet-start
  - add ip-addresses for localnet-setup generation
  - add localnet-setup folder to keep localnet-setup-artifacts
  - map localnet-setup-artifacts to containers
  - change start-docker.sh to pick up generated localnet-setup-artifacts

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Review by fedekunze — COMMENTED** [review]
Thanks! Looks like this only works if the repository is on the `$GOPATH`, can you fix that? Otherwise, LGTM

**Comment by simonpfeifhofer** [comment]
> Thanks! Looks like this only works if the repository is on the `$GOPATH`, can you fix that? Otherwise, LGTM

Thank you for the feedback. 
Really? Tested it with the repository __not__ on `$GOPATH`. Any hints for me to reproduce it? Happy to fix it.

**Comment by simonpfeifhofer** [comment]
@fedekunze would be happy to get additional info about your finding.

File: Makefile [hunk 1]
```diff
@@ -477,21 +477,21 @@ proto-update-deps:
...
-	IF not exist "build/node0/$(ETHERMINT_BINARY)/config/genesis.json" docker run --rm -v $(CURDIR)/build\ethermint\Z ethermintd/node "ethermintd testnet --v 4 -o /ethermint --keyring-backend=test"
+	IF not exist "build/node0/$(ETHERMINT_BINARY)/config/genesis.json" docker run --rm -v $(CURDIR)/build\ethermint\Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses 192.168.10.2,192.168.10.3,192.168.10.4,192.168.10.5"
 	docker-compose up -d
 else
-	mkdir -p ./build/
+	mkdir -p localnet-setup
 	@$(MAKE) docker-localnet
 
-	if ! [ -f build/node0/$(ETHERMINT_BINARY)/config/genesis.json ]; then docker run --rm -v $(CURDIR)/build:/ethermint:Z ethermintd/node "ethermintd testnet --v 4 -o /ethermint --keyring-backend=test"; fi
+	if ! [ -f localnet-setup/node0/$(ETHERMINT_BINARY)/config/genesis.json ]; then docker run --rm -v $(CURDIR)/localnet-setup:/ethermint:Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses 192.168.10.2,192.168.10.3,192.168.10.4,192.168.10.5"; fi
```
Comment (khoslaventures) [inline Makefile:494]: this is the output I got after running `make localnet-start`

First time:
```
if ! [ -f localnet-setup/node0/ethermintd/config/genesis.json ]; then docker run --rm -v /Users/akashkhosla/go/src/github.com/tharsis/ethermint/localnet-setup:/ethermint:Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses 192.168.10.2,192.168.10.3,192.168.10.4,192.168.10.5"; fi
Successfully initialized 4 node directories
docker-compose up -d
Docker Compose is now in the Docker CLI, try `docker compose up`

Creating network "ethermint_localnet" with driver "bridge"
ERROR: Pool overlaps with other one on this address space
make: *** [localnet-start] Error 1
```

Second time:
```
if ! [ -f localnet-setup/node0/ethermintd/config/genesis.json ]; then docker run --rm -v /Users/akashkhosla/go/src/github.com/tharsis/ethermint/localnet-setup:/ethermint:Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses 192.168.10.2,192.168.10.3,192.168.10.4,192.168.10.5"; fi
docker-compose up -d
Docker Compose is now in the Docker CLI, try `docker compose up`

Creating network "ethermint_localnet" with driver "bridge"
ERROR: Pool overlaps with other one on this address space
make: *** [localnet-start] Error 1
```

File: Makefile [hunk 2]
```diff
@@ -477,21 +477,21 @@ proto-update-deps:
...
 
 # Run a 4-node testnet locally
 localnet-start: localnet-stop
 ifeq ($(OS),Windows_NT)
-	mkdir build &
+	mkdir localnet-setup &
 	@$(MAKE) docker-localnet
 
-	IF not exist "build/node0/$(ETHERMINT_BINARY)/config/genesis.json" docker run --rm -v $(CURDIR)/build\ethermint\Z ethermintd/node "ethermintd testnet --v 4 -o /ethermint --keyring-backend=test"
+	IF not exist "build/node0/$(ETHERMINT_BINARY)/config/genesis.json" docker run --rm -v $(CURDIR)/build\ethermint\Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses 192.168.10.2,192.168.10.3,192.168.10.4,192.168.10.5"
```
[the remark this answers is not in the document]
Comment (fedekunze) [reply Makefile:488]: @leejw51crypto maybe?

File: Makefile (hunk 1 above)
Comment (simonpfeifhofer) [reply Makefile]: @khoslaventures thank you! Fixed with a6fa95f0fb42692588581829c55438c13934383d.

File: Makefile [hunk 3]
```diff
@@ -477,21 +477,21 @@ proto-update-deps:
 
 # Run a 4-node testnet locally
 build-docker-local-ethermint:
-	@$(MAKE) -C networks/local
+	@$(MAKE) docker-localnet
```
[the remark this answers is not in the document]
Comment (simonpfeifhofer) [reply Makefile]: Thank you! Done 9623f6455dee859ff6b20bc5c2d9f117c35362db. Fixed some additional issues.

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (simonpfeifhofer) [reply Makefile]: Parts for Windows are in addition affected by a6fa95f0fb42692588581829c55438c13934383d and 9623f6455dee859ff6b20bc5c2d9f117c35362db.

@khoslaventures @fedekunze: It's in general a must to support windows? It's an option for affected people to use WSL2?

[the remark this answers is not in the document]
Comment (fedekunze) [reply Makefile]: > It's in general a must to support Windows?

not really

**Comment by fedekunze** [comment]
having some issues running `make localnet-start`
```
make localnet-start
docker-compose down
Removing network ethermint_localnet
WARNING: Network ethermint_localnet not found.
mkdir -p localnet-setup
docker build --tag ethermintd/node ../.. -f ethermintnode/Dockerfile
[+] Building 5.6s (15/15) FINISHED                                                                                                                                                                                                 
 => [internal] load build definition from Dockerfile                                                                                                                                                                          0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                                             0.0s
 => => transferring context: 34B                                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/golang:1.14                                                                                                                                                                0.6s
 => [internal] load metadata for docker.io/library/golang:stretch                                                                                                                                                             0.6s
 => [build-env 1/5] FROM docker.io/library/golang:stretch@sha256:9be20fec15ebf0f28d16b03cd93a2152513da0e85af45e3f3d3381deab818bf5                                                                                             0.0s
 => [final 1/5] FROM docker.io/library/golang:1.14@sha256:1a7173b5b9a3af3e29a5837e0b2027e1c438fd1b83bbee8f221355087ad416d6                                                                                                    0.0s
 => [internal] load build context                                                                                                                                                                                             4.8s
 => => transferring context: 7.65MB                                                                                                                                                                                           4.6s
 => CACHED [final 2/5] RUN apt-get update                                                                                                                                                                                     0.0s
 => CACHED [build-env 2/5] RUN apt-get update && apt-get upgrade -y &&     apt-get install -y curl make git libc-dev bash gcc                                                                                                 0.0s
 => CACHED [build-env 3/5] WORKDIR /go/src/github.com/tharsis/ethermint                                                                                                                                                       0.0s
 => CACHED [build-env 4/5] COPY . .                                                                                                                                                                                           0.0s
 => CACHED [build-env 5/5] RUN make build-linux                                                                                                                                                                               0.0s
 => CACHED [final 3/5] COPY --from=build-env /go/src/github.com/tharsis/ethermint/build/ethermintd /                                                                                                                          0.0s
 => CACHED [final 4/5] COPY --from=build-env /go/src/github.com/tharsis/ethermint/scripts/start-docker.sh /                                                                                                                   0.0s
 => exporting to image                                                                                                                                                                                                        0.0s
 => => exporting layers                                                                                                                                                                                                       0.0s
 => => writing image sha256:9def4ee4087246481945ad537bf7977f1fa55bc66cada66c44c59792f5375373                                                                                                                                  0.0s
 => => naming to docker.io/ethermintd/node                                                                                                                                                                                    0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
if ! [ -f localnet-setup/node0/ethermintd/config/genesis.json ]; then docker run --rm -v /Users/federico/tharsis/ethermint/localnet-setup:/ethermint:Z ethermintd/node "./ethermintd testnet --v 4 -o /ethermint --keyring-backend=test --ip-addresses ethermintdnode0,ethermintdnode1,ethermintdnode2,ethermintdnode3"; fi
/bin/bash: ./ethermintd: cannot execute binary file: Exec format error
make: *** [localnet-start] Error 126
```

**Comment by simonpfeifhofer** [comment]
@fedekunze thank you! Added 6d52409934732704b4199db5557d839364e67810. Can you retry?

**Review by fedekunze — APPROVED** [review]
tested ACK 💯

**Comment by fedekunze** [comment]
@khoslaventures feel free to leave an additional review, or I can merge it by EOD, and then we can address your comments later on a follow-up PR 👍

**Comment by simonpfeifhofer** [comment]
@khoslaventures apart from the Windows portions I tried to address your points. IMO we should remove the Windows parts. There are additional-points (a few already visible as issues #339, #370) which are necessary to fix if Windows should be fully supported. @leejw51crypto what do you think about that since you have created the issues? Is it an option to use WSL2?

**Comment by fedekunze** [comment]
@simonpfeifhofer can you pull the latest changes? I enabled auto-merge

**Comment by simonpfeifhofer** [comment]
> @simonpfeifhofer can you pull the latest changes? I enabled auto-merge

@fedekunze done.

**Comment by simonpfeifhofer** [comment]
@fedekunze please let me know if there is still something which is missing in your opinion.

**Comment by fedekunze** [comment]
@simonpfeifhofer I enabled auto-merge but it seems that you need to update your configuration as an external contributor to accept upstream rebasing (when the branch needs to be updated)

**Comment by fedekunze** [comment]
@simonpfeifhofer I will add some changes to the contributor guidelines to prevent this from happening again. I was referring to this doc to allow changes from maintainers:

https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request merged**